src/utils/format_utils.cxx

```diff
@@ -242,6 +242,58 @@ string string_to_hex(const string &arg)
     return ret;
 }
 
+static uint8_t get_nibble(char b)
+{
+    if ('0' <= b && b <= '9')
+    {
+        return b - '0';
+    }
+    if ('a' <= b && b <= 'f')
+    {
+        return b - 'a' + 10;
+    }
+    if ('A' <= b && b <= 'F')
+    {
+        return b - 'A' + 10;
+    }
+    return 0xff;
+}
+
+ssize_t hex_to_string(
```

Comment:
I don't see any instance where the return value may be negative. Should we make the return type size_t and avoid any need to cast the "len" type?

Reply:
done